apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/summarize-existing-entities.ts

```diff
@@ -0,0 +1,146 @@
+import type { VersionedUrl } from "@blockprotocol/type-system";
+import type { Entity, EntityId } from "@local/hash-subgraph";
+import dedent from "dedent";
+
+import { getFlowContext } from "../../shared/get-flow-context";
+import { getLlmResponse } from "../../shared/get-llm-response";
+import { getToolCallsFromLlmAssistantMessage } from "../../shared/get-llm-response/llm-message";
+import type { LlmToolDefinition } from "../../shared/get-llm-response/types";
+import { graphApiClient } from "../../shared/graph-api-client";
+import { simplifyEntity } from "../../shared/simplify-entity";
+
+export type ExistingEntitySummary = {
+  entityId: EntityId;
+  entityTypeId: VersionedUrl;
+  name: string;
+  summary: string;
+};
+
+const systemPrompt = dedent(`
+  You are an entity summary agent.
+
+  The user will provide you with:
+    - "entities": a list of entities for which you need to provide a "name" and a "summary".
+
+  You must provide a summary for every entity in the list provided by the user.
+`);
+
+const toolName = "submitEntitySummaries";
+
+const registerEntitySummariesToolDefinition: LlmToolDefinition<
+  typeof toolName
+> = {
+  name: toolName,
+  description: "Register entity summaries",
+  inputSchema: {
+    type: "object",
+    properties: {
+      entitySummaries: {
+        type: "array",
+        items: {
+          type: "object",
+          properties: {
+            entityId: {
+              type: "string",
+              description: dedent(`
+                The "name" should be as unique as possible based on the properties of the entity.
+                For example for an entity of type "Person" with first name "John", middle name "Doe", and last name "Smith", the "name" could be "John Doe Smith".
```

Comment:
What should the description be here?

Reply:
Apologies, this should be the description for the identifier no the `"name"` field. Have addressed in https://github.com/hashintel/hash/pull/4497/commits/7b31692fcc6fa3beaa1e7eca945d951b91727949